ui/components/app/confirm/info/row/currency.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+
+import {
+  AlignItems,
+  Display,
+  FlexWrap,
+} from '../../../../../helpers/constants/design-system';
+import { Box } from '../../../../component-library';
+import UserPreferencedCurrencyDisplay from '../../../user-preferenced-currency-display/user-preferenced-currency-display.component';
+import CurrencyDisplay from '../../../../ui/currency-display/currency-display.component';
+
+export type ConfirmInfoRowCurrencyProps = {
+  value: number | string;
+  currency?: string;
+};
+
+// todo: the component currently takes care of displaying value in the currency passed
+// it will default to using User's preferred currency.
+// As we encounter different use cases in future we would need to extend this
+// to support more configurations / formatting options.
+export const ConfirmInfoRowCurrency = ({
+  value,
+  currency,
+}: ConfirmInfoRowCurrencyProps) => (
+  <Box
+    display={Display.Flex}
+    alignItems={AlignItems.center}
+    flexWrap={FlexWrap.Wrap}
+    style={{
+      columnGap: '8px',
+      fontSize: 'var(--font-size-3)',
+    }}
+  >
+    {currency ? (
+      <CurrencyDisplay
+        currency={currency}
+        value={value ? `${value}` : undefined}
```

Comment:
is there a use case for an undefined value? if yes, since it's currency should we display 0.00 instead?

Reply:
Oh this was a good point. `value` is mandatory and typed as [`string | number`](https://github.com/MetaMask/metamask-extension/pull/23260/files#diff-f5b2f432afcf046ad259498c46bc5c67bac8fc75964c7d5b3d80c2eeea623b2cR13), so why are we checking if it is defined?

Can't we just:
```js
value={`${value}`}
```

Reply:
I updated value to `value={`${value}`}`